walletunlocker/service.go

```diff
@@ -306,9 +336,25 @@ func (u *UnlockerService) InitWallet(ctx context.Context,
 		initMsg.ChanBackups = *chansToRestore
 	}
 
-	u.InitMsgs <- initMsg
+	// Deliver the initialization message back to the main daemon.
+	select {
+	case u.InitMsgs <- initMsg:
+		// We need to read from the channel to let the daemon continue
+		// its work and to get the admin macaroon. Once the response
+		// arrives, we directly forward it to the client.
+		select {
+		case adminMac := <-u.MacResponseChan:
```

Comment:
This series of sends looks a bit tricky....but they do indeed appear sound. Wasn't able to trigger a deadlock or anything locally.

Reply:
Yeah, it took us a few iterations to get rid of all the deadlocks.